python/ray/services.py

```diff
@@ -393,81 +393,15 @@ def check_version_info(redis_client):
             print(error_message)
 
 
-def start_credis(node_ip_address,
-                 redis_address,
-                 port=None,
-                 redirect_output=False,
-                 cleanup=True):
-    """Start the credis global state store.
-
-    Credis is a chain replicated reliable redis store. It consists
-    of one master process that acts as a controller and a number of
-    chain members (currently two, the head and the tail).
-
-    Args:
-        node_ip_address: The IP address of the current node. This is only used
-            for recording the log filenames in Redis.
-        redis_address (str): The IP address and port of the primary redis
-            server.
-        port (int): If provided, the primary Redis shard will be started on
-            this port.
-        redirect_output (bool): True if output should be redirected to a file
-            and false otherwise.
-        cleanup (bool): True if using Ray in local mode. If cleanup is true,
-            then all Redis processes started by this method will be killed by
-            services.cleanup() when the Python process that imported services
-            exits.
-
-    Returns:
-        The address (ip_address:port) of the credis master process.
-    """
-
-    components = ["credis_master", "credis_head", "credis_tail"]
-    modules = [
-        CREDIS_MASTER_MODULE, CREDIS_MEMBER_MODULE, CREDIS_MEMBER_MODULE
-    ]
-    ports = []
-
-    for i, component in enumerate(components):
-        stdout_file, stderr_file = new_log_files(component, redirect_output)
-
-        new_port, _ = start_redis_instance(
-            node_ip_address=node_ip_address,
-            port=port,
-            stdout_file=stdout_file,
-            stderr_file=stderr_file,
-            cleanup=cleanup,
-            module=modules[i],
-            executable=CREDIS_EXECUTABLE)
-
-        ports.append(new_port)
-
-    [master_port, head_port, tail_port] = ports
-
-    # Connect the members to the master
-
-    master_client = redis.StrictRedis(host=node_ip_address, port=master_port)
-    master_client.execute_command("MASTER.ADD", node_ip_address, head_port)
-    master_client.execute_command("MASTER.ADD", node_ip_address, tail_port)
-
-    credis_address = address(node_ip_address, master_port)
-
-    # Register credis master in redis
-    redis_ip_address, redis_port = redis_address.split(":")
-    redis_client = redis.StrictRedis(host=redis_ip_address, port=redis_port)
-    redis_client.set("credis_address", credis_address)
-
-    return credis_address
-
-
 def start_redis(node_ip_address,
                 port=None,
                 redis_shard_ports=None,
                 num_redis_shards=1,
                 redis_max_clients=None,
                 redirect_output=False,
                 redirect_worker_output=False,
-                cleanup=True):
+                cleanup=True,
+                use_credis=("RAY_USE_NEW_GCS" in os.environ)):
```

Comment:
This is not a good idea in Python since this statement `("RAY_USE_NEW_GCS" in os.environ)` is only evaluated once, when the function is defined. Better to leave the default as `None` and then check the condition in the function body.

Reply:
Done.